components/browser/engine-gecko-beta/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt

```diff
@@ -359,11 +359,13 @@ class GeckoEngineSession(
 
         override fun onExternalResponse(session: GeckoSession, response: GeckoSession.WebResponseInfo) {
             notifyObservers {
-                onExternalResource(
-                    url = response.uri,
-                    contentLength = response.contentLength,
-                    contentType = response.contentType,
-                    fileName = response.filename)
+                response.filename?.let {
+                    onExternalResource(
+                            url = response.uri,
+                            contentLength = response.contentLength,
+                            contentType = response.contentType,
+                            fileName = it)
+                }
```

Comment:
⚠️ We **always** want to invoke the callback! If filename is null then please use `DownloadUtils.guessFileName()`.